MH4U Dex/CoreDataController.m

```diff
@@ -46,88 +48,79 @@ + (CoreDataController *)sharedCDController
-            MonsterDrop *newDrop = [MonsterDrop insertInManagedObjectContext:self.managedObjectContext];
-            newDrop.id = drop[MHDMonsterDropIDKey];
-            newDrop.method = drop[MHDMonsterDropMethodKey];
-            newDrop.quantity = drop[MHDMonsterDropQuantityKey];
-            
-            // If the Item for this drop already exists in the persistent store ...
-            NSPredicate *itemPredicate = [NSPredicate predicateWithFormat:@"%K == %@", ItemAttributes.name, drop[MHDMonsterDropItemNameKey]];
-            Item *item = (Item *)[self uniqueEntityWithEntityName:[Item entityName] withPredicate:itemPredicate];
-            if (!item) {
-                item = [Item insertInManagedObjectContext:self.managedObjectContext];
-                item.name = drop[MHDMonsterDropItemNameKey];
+    //Repeat this process for ten different JSON files
+    for (NSUInteger fileNumber = 1; fileNumber <= 10; fileNumber++) {
+        NSLog(@"Loading in monster drop file # %@", @(fileNumber));
+        NSArray *dropList = [self loadArrayFromJsonFileNamed:[MHDMonsterDropsFileName stringByAppendingFormat:@"%@", @(fileNumber)]];
+        for (NSDictionary *monsterDropDict in dropList) {
+            //Check if the monster exists.  Do not bother even creating the drop if it does not
+            Monster *monster = [self monsterWithName:monsterDropDict[MHDMonsterDropMonsterNameKey]];
+            if (!monster) {
+                continue;
             }
-            newDrop.item = item;
-            // If this is a monster drop ...
-            NSString *monsterName = drop[MHDMonsterDropMonsterNameKey];
-            if (monsterName.length) {
-                // We need to add this drop to the monster's list, if it is not there already.
-                
-                // First we should ensure that the monster already exists in the persistent store
-                NSNumber *monsterID = drop[MHDMonsterDropMonsterIDKey];
-                NSPredicate *monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %d", MonsterAttributes.id, monsterID.intValue];
-                Monster *monster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName] withPredicate:monsterPredicate];
-                if (!monster) {
-                    // TODO: Handle better, later.
-                    NSLog(@"The Monster, %@, was not found!", monsterName);
-                } else {
-                    newDrop.monsterSource = monster;
-                    newDrop.rank = drop[MHDMonsterDropRankKey];
-                    newDrop.percent = drop[MHDMonsterDropPercentKey];
-                }
+            NSPredicate *monsterDropPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterDropAttributes.id, monsterDropDict[MHDMonsterDropIDKey]];
+            MonsterDrop *monsterDrop = (MonsterDrop *)[self uniqueEntityWithEntityName:[MonsterDrop entityName] withPredicate:monsterDropPredicate];
```

Comment:
Consider re-breaking these two lines so they aren't as long (lines should generally be 120 characters or fewer).